util/system/path.go

```diff
@@ -16,3 +25,202 @@ func DefaultPathEnv(os string) string {
 	}
 	return DefaultPathEnvUnix
 }
+
+// NormalizePath cleans the path based on the operating system the path is meant for.
+// It takes into account a potential parent path, and will join the path to the parent
+// if the path is relative. Additionally, it will apply the folliwing rules:
+//   - always return an absolute path
+//   - always strip drive letters for Windows paths
+//   - optionally keep the trailing slashes on paths
+//   - paths are returned using forward slashes
+func NormalizePath(parent, newPath, inputOS string, keepSlash bool) (string, error) {
+	if inputOS == "" {
```

Comment:
Do we need to make os optional? Can't the caller just always pass it?

Reply:
I can have the functions err out if it's empty if you prefer.

Reply:
Was not sure what the preference was :smile: .

Reply:
Fair enough. I set ```linux``` as the default.